core/src/main/java/com/oath/oak/Rebalancer.java

```diff
@@ -177,14 +178,16 @@ RebalanceResult createNewChunks() {
 
                 List<Chunk<K,V>> frozenSuffix = frozenChunks.subList(iterFrozen.previousIndex(), frozenChunks.size());
                 // try to look ahead and add frozen suffix
-                if (canAppendSuffix(frozenSuffix, maxRangeToAppend, maxBytesToAppend)) {
+                if (canAppendSuffix(frozenSuffix, maxRangeToAppend)) {
                     // maybe there is just a little bit copying left
                     // and we don't want to open a whole new chunk just for it
                     completeCopy(currNewChunk, ei, frozenSuffix);
                     break;
                 } else {
                     // we have to open an new chunk
-                    // TODO do we want to use slice here?
+                    // here we create a new on-heap minimal key for the second new chunk,
+                    // created by the split. The new min key is on-heap copy of the one off-heap
+                    // TODO do we want to use slice here? yes we need!
```

Comment:
Suggest to remove the TODO and instead explain why we need slice here

Reply:
done